sw/host/opentitantool/src/command/image.rs

```diff
@@ -160,77 +168,54 @@ impl CommandDispatch for ManifestUpdateCommand {
             }
         }
 
-        // A u8 buffer with u32 alignment to be able to deserialize bytes for extensions.
-        // Note: This should ideally be based on the alignment requirements of the extensions but
-        // just using `u32` works since all extensions we have are u32-aligned. We can't miss If
-        // this doesn't hold since `LayoutVerified::new()` errors out.
-        #[repr(C)]
-        pub struct Buffer32 {
-            pub bytes: [u8; image::Image::MAX_SIZE],
-            _align: [u32; 0],
-        }
-        let mut buf = Buffer32 {
-            bytes: [0; image::Image::MAX_SIZE],
-            _align: [],
-        };
+        let ext = self
+            .manifest_ext
+            .as_deref()
+            .map(ManifestExtSpec::read_from_file)
+            .unwrap_or(Ok(Default::default()))?;
 
-        // Update `signed_region_end` if it is 0, otherwise it has already been set (i.e., to
-        // support offline signing scenarios).
-        if image.borrow_manifest()?.signed_region_end == 0 {
-            image.update_signed_region_end()?;
-        }
+        // Add any signed extensions defined in `ext`.
+        image.add_signed_manifest_extensions(&ext)?;
 
         let mut spx_private_key: Option<SpxKeypair> = None;
         if let Some(key) = &self.spx_key {
+            ext.check_for_duplicates(ManifestExtId::spx_key)?;
```

Comment:
nit: I think it would be safer to move this check to the methods that can introduce duplicates, e.g. `add_manifest_extension`. Also, can you please add a comment explaining why we need this check? Is it because of the _raw_ extensions in the HJSON file?